src/main/java/org/apache/gossip/manager/GossipCore.java

```diff
@@ -72,11 +55,12 @@
   private final Meter messageSerdeException;
   private final Meter tranmissionException;
   private final Meter tranmissionSuccess;
-  
+  private final Map<Class, MessageHandler> handlers;
```

Comment:
Please put this inside GossipManager. I understand why you put it here, but we are moving away from GossipCore because of all the method chaining GossipService->GossipManager->GossipCore. Also should we use ConcurrentMap so that if handlers are added at runtime they will be seen?